swarm/api/routes/wisdom.py

```
@@ -165,6 +167,17 @@ def _get_runs_root() -> Path:
     """Get the wisdom directory for a run."""
+    try:
+        run_id = validate_path_component(run_id, "run_id")
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "invalid_path",
+                "message": str(e),
+                "details": {"run_id": run_id},
+            },
+        )
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)